src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -146,7 +152,8 @@ public void testFromXContent() throws IOException {
     @Test
     public void testToQuery() throws IOException {
         QueryParseContext context = new QueryParseContext(index, queryParserService);
-        assertLuceneQuery(this.testQuery, this.testQuery.toQuery(context));
+        context.setAllowUnmappedFields(true);
```

Comment:
I had thought this was a singlenodetest, my mistake! But it does look like you are creating an IndexQueryParserService, which has a MapperService, yet I don't see it in the initialized injection? 

That said, you can fake this out with a test QueryParserContext? That is what each query parser calls to get the mapping for a field. Otherwise always `unmappedFieldAsString` you would always get a string field, but this won't work for other query parsers right?